service.sleepy.watchdog/default.py

```diff
@@ -0,0 +1,293 @@
+            xbmc.Player().stop()
+            if self.jumpMainMenu:
+                xbmc.sleep(500)
+                notifyLog('jump to main menu')
+                xbmc.executebuiltin('ActivateWindow(home)')
+
+    @classmethod
+    def quit(cls):
+        notifyLog('quit kodi')
+        xbmc.executebuiltin('Quit')
+
+    @classmethod
+    def systemReboot(cls):
+        notifyLog('init system reboot')
+        xbmc.restart()
+
+    @classmethod
+    def systemShutdown(cls):
+        notifyLog('init system shutdown')
+        xbmc.shutdown()
+
+    @classmethod
+    def systemHibernate(cls):
+        notifyLog('init system hibernate')
+        xbmc.executebuiltin('Hibernate')
+
+    @classmethod
+    def systemSuspend(cls):
+        notifyLog('init system suspend')
+        xbmc.executebuiltin('Suspend')
+
+    def sendCecCommand(self):
+        if not self.sendCEC: return
+        notifyLog('send standby command via CEC')
+        cec = subprocess.Popen('echo "standby 0" | cec-client -s', stdout=subprocess.PIPE, shell=True).communicate()
+        for retstr in cec: notifyLog(str(retstr).strip())
+
+    def runAddon(self):
+        if xbmc.getCondVisibility('System.HasAddon(%s)' % (self.addon_id.split(',')[0])):
+            notifyLog('run addon \'%s\'' % (self.addon_id))
+            xbmc.executebuiltin('RunAddon(%s)' % (self.addon_id))
+        else:
+            notifyLog('could not run nonexistent addon \'%s\'' % (self.addon_id.split(',')[0]), level=xbmc.LOGERROR)
+
+    def start(self):
+
+        _currentIdleTime = 0
+        _maxIdleTime = self.maxIdleTime
+        _msgCnt = 0
+
+        while not xbmc.Monitor.abortRequested(self):
+            self.actionCanceled = False
+
+            if _msgCnt % 10 == 0 and _currentIdleTime > 60 and not self.testConfig:
+                notifyLog('idle time in active time frame: %s' % (time.strftime('%H:%M:%S', time.gmtime(_currentIdleTime))))
+
+            if _currentIdleTime > xbmc.getGlobalIdleTime():
+                notifyLog('user activity detected, reset idle time')
+                _msgCnt = 0
+                _maxIdleTime = self.maxIdleTime
+                _currentIdleTime = 0
+
+            _msgCnt += 1
+
+            # Check if GlobalIdle longer than maxIdle and we're in a time frame
+
+            if self.activeTimeFrame(debug=True):
+                if _currentIdleTime > (_maxIdleTime - int(self.notifyUser)*self.notificationTime):
+
+                    notifyLog('max idle time reached, ready to perform some action')
+
+                    # Check if notification is allowed
+                    if self.notifyUser:
+                        count = 0
+                        notifyLog('init notification countdown for action no. %s' % (self.action))
+
+                        while (self.notificationTime - count > 0):
+                            notifyUser(__LS__(32115) % (__LS__(self.action), self.notificationTime - count))
+                            xbmc.sleep(9000)
```

Comment:
`xbmc.sleep` is something discourage people from using especially if the sleep time is more than just a few ms.
Please use `xbmc.Monitor.waitForAbort` instead